cairo/protocol_handler/src/protocol_handler.cairo

```diff
@@ -386,5 +437,69 @@ pub mod ProtocolHandler {
+        //*                             SELF MANAGEMENT                              *//
+        //* ------------------------------------------------------------------------ *//
+
+        fn change_operator(ref self: ContractState, new_operator_address: ContractAddress) {
+            // Check only security council can call
+            self.accesscontrol.assert_only_role(SECURITY_COUNCIL_ROLE);
+
+            // Revoke the OPERATOR_ROLE from the current operator
+            self.accesscontrol._revoke_role(OPERATOR_ROLE, self.operator.read());
+
+            // Grant role to the new operator
+            self.accesscontrol._grant_role(OPERATOR_ROLE, new_operator_address);
+
+            // Update the operator
+            self.operator.write(new_operator_address);
+        }
+
+        fn change_security_council(
+            ref self: ContractState, new_security_council_address: ContractAddress
+        ) {
+            // Check only security council can call
+            self.accesscontrol.assert_only_role(SECURITY_COUNCIL_ROLE);
+
+            // Revoke the SECURITY_COUNCIL_ROLE from the current security council
+            self.accesscontrol._revoke_role(SECURITY_COUNCIL_ROLE, self.security_council.read());
+
+            // Grant role to the new security council
+            self.accesscontrol._grant_role(SECURITY_COUNCIL_ROLE, new_security_council_address);
+
+            // Update the security council
+            self.security_council.write(new_security_council_address);
+        }
+
+        fn change_gas_price_admin(ref self: ContractState, new_gas_price_admin: ContractAddress) {
+            // Check only security council can call
+            self.accesscontrol.assert_only_role(SECURITY_COUNCIL_ROLE);
+
+            // Revoke the GAS_PRICE_ADMIN_ROLE from the current gas price admin
+            self.accesscontrol._revoke_role(GAS_PRICE_ADMIN_ROLE, self.gas_price_admin.read());
+
+            // Grant role to the new gas price admin
+            self.accesscontrol._grant_role(GAS_PRICE_ADMIN_ROLE, new_gas_price_admin);
+
+            // Update the gas price admin
+            self.gas_price_admin.write(new_gas_price_admin);
+        }
+
+        fn add_guardian(ref self: ContractState, new_guardian_address: ContractAddress) {
+            // Check only security council can call
+            self.accesscontrol.assert_only_role(SECURITY_COUNCIL_ROLE);
+
+            // Grant the GUARDIAN_ROLE to the new guardian
+            self.accesscontrol._grant_role(GUARDIAN_ROLE, new_guardian_address);
+        }
+
+        fn remove_guardian(ref self: ContractState, guardian_address: ContractAddress) {
+            // Check only security council can call
+            self.accesscontrol.assert_only_role(SECURITY_COUNCIL_ROLE);
+
+            // Revoke the GUARDIAN_ROLE from the guardian
+            self.accesscontrol._revoke_role(GUARDIAN_ROLE, guardian_address);
+        }
```

Comment:
After reflexion not sure the guardians storage variable is necessary ?
It is not used 
For other storage variables, we use them for the revocation to avoid submitting them with the new one. 
Here, it is a removal or addition of a specific address and it is already handled by the AccessControl.

I think we could remove this storage variable

Reply:
ok